src/orange/orangeinp/ConvexRegion.cc

```diff
@@ -368,6 +368,75 @@ GenTrap::GenTrap(real_type halfz, VecReal2 const& lo, VecReal2 const& hi)
     }
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a TRD from 5 half-lengths: one half-height, {hx1,hy1} for a in -z,
+ * and {hx2,hy2} in +z
+ */
+GenTrap GenTrap::from_trd(real_type halfz, Real2 const& lo, Real2 const& hi)
+{
+    CELER_VALIDATE(lo[0] > 0, << "nonpositive lower x half-edge: " << lo[0]);
+    CELER_VALIDATE(hi[0] > 0, << "nonpositive upper x half-edge: " << hi[0]);
+    CELER_VALIDATE(lo[1] > 0, << "nonpositive lower y half-edge: " << lo[1]);
+    CELER_VALIDATE(hi[1] > 0, << "nonpositive upper y half-edge: " << hi[1]);
+    CELER_VALIDATE(halfz > 0, << "nonpositive half-height: " << halfz);
+
+    VecReal2 lower
+        = {{-lo[0], -lo[1]}, {lo[0], -lo[1]}, {lo[0], lo[1]}, {-lo[0], lo[1]}};
+    VecReal2 upper
+        = {{-hi[0], -hi[1]}, {hi[0], -hi[1]}, {hi[0], hi[1]}, {-hi[0], hi[1]}};
+
+    return std::move(GenTrap{halfz, std::move(lower), std::move(upper)});
```

Comment:
To fix the build error (you don't need to `move` when returning a local variable or constructing an rvalue):

```suggestion
    return GenTrap{halfz, std::move(lower), std::move(upper)};
```